plugins/woocommerce/client/blocks/assets/js/blocks/product-details/edit.tsx

```diff
@@ -109,24 +45,115 @@ const useIsInvalidQueryLoopContext = ( clientId: string, postType: string ) => {
 const Edit = ( { clientId, context }: ProductDetailsEditProps ) => {
 	const blockProps = useBlockProps();
 
+	const product = useSelect(
+		( select ) => {
+			if ( ! context.postId ) {
+				return null;
+			}
+			const { getProduct } = select( productsStore );
+			return getProduct( Number( context.postId ) );
+		},
+		[ context.postId ]
+	);
+
+	const isInnerBlockOfSingleProductBlock = useSelect(
+		( select ) => {
+			const blocks = select(
+				blockEditorStore
+				// @ts-expect-error - getBlockParentsByBlockName is not typed
+			).getBlockParentsByBlockName(
+				clientId,
+				'woocommerce/single-product'
+			);
+			return blocks.length > 0;
+		},
+		[ clientId ]
+	);
+
+	const template = useMemo( () => {
+		return getTemplate( product, {
+			isInnerBlockOfSingleProductBlock,
+		} );
+	}, [ product, isInnerBlockOfSingleProductBlock ] );
+
 	const { hasInnerBlocks, wasBlockJustInserted } = useSelect(
 		( select ) => {
+			// @ts-expect-error - getBlocks is not typed
 			const blocks = select( blockEditorStore ).getBlocks( clientId );
+			const innerBlocks = blocks.length > 0;
```

Comment:
Nitpick: since it's a boolean, not actual innerBlocks, what about renaming it:

```suggestion
			const hasInnerBlocks = blocks.length > 0;
```

EDIT: Ok, I see what happens here, if you name it `has...` there will be naming conflict, the same goes for `wasBlockJust...` vs `blockJust...` 😂 

Keep it as-is or consider simply returning object with no const assingments:

```
{
  hasInnerBlocks: blocks.length > 0,
  wasBlockJustInserted: select( blockEditorStore ).wasBlockJustInserted( clientId ),
}